sql/table.go

```diff
@@ -300,29 +300,36 @@ func encodeSecondaryIndexes(tableID structured.ID, indexes []structured.IndexDes
 	return secondaryIndexEntries, nil
 }
 
-// TODO(tamird): make this not panic. Not critical, since a panic here
-// will just tank a single goroutine on the server and be silently
-// swallowed.
-func prepareVal(col structured.ColumnDescriptor, val parser.Expr) (interface{}, error) {
+func prepareVal(col structured.ColumnDescriptor, val parser.Datum) (interface{}, error) {
 	if val == parser.DNull {
 		return nil, nil
 	}
 
 	switch col.Type.Kind {
 	case structured.ColumnType_BIT:
-		return bool(val.(parser.DBool)), nil
+		if v, ok := val.(parser.DBool); ok {
```

Comment:
Just noticed that this was incorrect before your change. `BIT` types should be stored as ints. Mind fleshing out the test cases.